app/models/directory_item.rb

```diff
@@ -163,6 +147,27 @@ def self.refresh_period!(period_type, force: false)
         SQL
     end
   end
+
+  def self.add_missing_users_all_periods
+    period_types.each_key { |p| add_missing_users(p) }
+  end
+
+  def self.add_missing_users(period_type)
+    column_names = DirectoryColumn.automatic_column_names + DirectoryColumn.plugin_directory_columns
+    DB.exec(
+      "INSERT INTO directory_items(period_type, user_id, #{column_names.map(&:to_s).join(", ")})
+              SELECT
+                  :period_type,
+                  u.id,
+                  #{Array.new(column_names.count) { |_| 0 }.join(", ")}
```

Comment:
You don't need the argument in the block.

```suggestion
                  #{Array.new(column_names.count) { 0 }.join(", ")}
```

You can alternatively do

```suggestion
                  #{Array.new(column_names.count, 0).join(", ")}
```